docs/tutorials/build-your-first-blazor-app.md

```diff
@@ -0,0 +1,482 @@
+
+![Blazor app Home page](https://msdnshared.blob.core.windows.net/media/2018/03/blazor-home.png)
+
+## Build components
+
+1. Browse to each of the app's three pages: Home, Counter, and Fetch data.
+
+    These three pages are implemented by the three Razor files in the *Pages* folder: *Index.cshtml*, *Counter.cshtml*, and *FetchData.cshtml*. Expand the *Pages* folder in the Solution Explorer to see these files. Each of these files implements a Blazor component that's compiled and executed client-side in the browser.
+    
+    ![Pages folder](https://user-images.githubusercontent.com/1874516/38259574-392a4e5e-371a-11e8-8b6b-66af4a6d07b6.png)
+
+1. Select the button on the Counter page.
+
+    ![Blazor app Home page](https://msdnshared.blob.core.windows.net/media/2018/03/blazor-counter.png)
+
+    Each time the button is selected, the counter is incremented without a page refresh. Normally, this kind of client-side behavior is handled in JavaScript; but here, it's implemented in C# and .NET by the `Counter` component.
+    
+1. Take a look at the implementation of the `Counter` component in the *Counter.cshtml* file:
+
+    ```cshtml
+    @page "/counter"
+
+    <h1>Counter</h1>
+
+    <p>Current count: @currentCount</p>
+
+    <button onclick="@IncrementCount">Click me</button>
+
+    @functions {
+        int currentCount = 0;
+
+        void IncrementCount()
+        {
+            currentCount++;
+        }
+    }
+    ```
+
+    The UI for the `Counter` component is defined using normal HTML. Dynamic rendering logic (for example, loops, conditionals, expressions) is added using an embedded C# syntax called [Razor](https://docs.microsoft.com/aspnet/core/mvc/views/razor). The HTML markup and C# rendering logic are converted into a component class at build time. The name of the generated .NET class matches the name of the file.
+
+    Members of the component class are defined in a `@functions` block. In the `@functions` block, component state (properties, fields) and methods can be specified for event handling or for defining other component logic. These members can then be used as part of the component's rendering logic and for handling events.
+
+    When the button is selected, the `Counter` component's registered `onclick` handler is called (the `IncrementCount` method) and the `Counter` component regenerates its render tree. Blazor compares the new render tree against the previous one and applies any modifications to the browser Document Object Model (DOM). The displayed count is updated.
+
+1. Update the markup for the `Counter` component to make the top-level header more *exciting*.
+
+    ```cshtml
+    @page "/counter"
+    <h1><em>Counter!!</em></h1>
+    ```
+
+1. Also modify the C# logic of the `Counter` component to make the count increment by two instead of one.
+
+    ```cshtml
+    @functions {
+        int currentCount = 0;
+
+        void IncrementCount()
+        {
+            currentCount += 2;
```

Comment:
Ideally we would highlight the lines of code in the snippets to indicate where the changes are. I think you can do that if we convert the code blocks to code snippets. But maybe that isn't worth the work since we're going to replace this tutorial anyway.

Reply:
Ideally, I agree. If it were as simple as just linking the sample files now, I'd just do it. However, the downloadable sample is the final finished app. These in-between step example code blocks would need to be loaded into their own files into a folder and linked from there. It is a bit much with all the 2.1 stuff on me at the moment. 😅

Reply:
Yup, let's leave it as is then.